### PR TITLE
Pipline module document template update

### PR DESCRIPTION
**Pull request by jonathansick**
- The template's README includes the document creation procedure based on the Slack note.
- In the _templated_ README:
  - Adjust the format of the README subtitle and description
  - Adapt the Difficulty badge's colour based on the difficulty itself
  - Add a Links section
- Add links in the tex file to spherex-tex where they are likely to be needed

### CODE REVIEW

File: pipeline-module/README.md [hunk 1]
```diff
@@ -1,3 +1,131 @@
...
+
+### 0. Prerequisites
+
+On your computer, ensure that [Cookiecutter](https://cookiecutter.readthedocs.io/en/latest/) is installed:
+
+```shell
+pip install -U cookiecutter
+```
+
+You also need Git >=
```
Comment (gpdf) [inline pipeline-module/README.md:16]: Is there a version number missing from this line?

File: pipeline-module/README.md [hunk 2]
```diff
@@ -1,3 +1,131 @@
...
+
+[Learn more about the spherex document class and spherex bibliography file.](https://github.com/SPHEREx/spherex-tex)
+
+Copy any images from the original document on Overleaf into the new repo.
+
+If the original document used `\modulename` in the body of the document to substitute in the name of the module, you will need to use `\themodulename` for this instead.
+(`\modulename{name}` is now the setter for the name, rather than the getter.)
+
+If you'd like, you can compile the document locally.
+See the [spherex-tex](https://github.com/SPHEREx/spherex-tex) repository for instructions.
```
Comment (gpdf) [inline pipeline-module/README.md:111]: "... repository's README for instructions."?

**Review by gpdf — COMMENTED** [review]
Looks good - a couple of suggestions.

File: pipeline-module/README.md (hunk 1 above)
Comment (jonathansick) [reply pipeline-module/README.md:16]: Ah, I was trying to find when the -M flag was added to git, but could never find it. I'll change the wording on this.

File: pipeline-module/README.md [hunk 3]
```diff
@@ -1,3 +1,131 @@
...
+
+[Learn more about the spherex document class and spherex bibliography file.](https://github.com/SPHEREx/spherex-tex)
+
+Copy any images from the original document on Overleaf into the new repo.
+
+If the original document used `\modulename` in the body of the document to substitute in the name of the module, you will need to use `\themodulename` for this instead.
+(`\modulename{name}` is now the setter for the name, rather than the getter.)
+
+If you'd like, you can compile the document locally.
+See the [spherex-tex](https://github.com/SPHEREx/spherex-tex) repository's for instructions.
```
Comment (gpdf) [inline pipeline-module/README.md:111]: See the [spherex-tex](https://github.com/SPHEREx/spherex-tex) repository's [README](https://github.com/SPHEREx/spherex-tex/blob/main/README.md) for instructions.

... was what I had in mind.

**Review by gpdf — CHANGES_REQUESTED** [review]
One last tweak

File: pipeline-module/README.md (hunk 3 above)
Comment (jonathansick) [reply pipeline-module/README.md]: Sorry about that, got the README link in now.

**Review by gpdf — APPROVED** [review]

**Pull request merged**